### PR TITLE
Audio URLs

### PR DESCRIPTION
**Pull request by regulad**
This adds support for Android IP webcam's Audio URLs. I plan to use this change to enable audio in the Home Assistant integration.

### CODE REVIEW

**Comment by MartinHjelmare** [comment]
How are you planning to use this in Home Assistant?

**Comment by regulad** [comment]
Add streaming support to the Android ip webcam and use ffmpeg to add the AAC stream, a proof of concept is here: https://github.com/regulad/hass-android_ip_webcam-audio/blob/master/custom_components/android_ip_webcam/camera.py

**Comment by MartinHjelmare** [comment]
I don't think we'll accept that code in the core integration as it has too much AV details and also socket code. That kind of code should be done in a 3rd party library. In a custom integration you can of course do what you want.

Do you still want to pursue this PR?

**Comment by regulad** [comment]
Yes, of course. That is only a proof of concept, I will rework it once this PR is complete and the audio is added to the backend library.

File: pydroid_ipcam/__init__.py [hunk 1]
```diff
@@ -48,6 +48,21 @@ def mjpeg_url(self) -> str:
         """Return mjpeg url."""
         return f"{self.base_url}/video"
 
+    @property
+    def audio_wav_url(self) -> str:
+        """Returns url that Waveform audio can be streamed from."""
```
Comment (MartinHjelmare) [inline pydroid_ipcam/__init__.py:53]:
```suggestion
        """Return url that Waveform audio can be streamed from."""
```

File: pydroid_ipcam/__init__.py [hunk 2]
```diff
@@ -48,6 +48,21 @@ def mjpeg_url(self) -> str:
...
+        return f"{self.base_url}/audio.wav"
+
+    @property
+    def audio_aac_url(self) -> str:
+        """Returns url that AAC audio can be streamed from."""
```
Comment (MartinHjelmare) [inline pydroid_ipcam/__init__.py:58]:
```suggestion
        """Return url that AAC audio can be streamed from."""
```

File: pydroid_ipcam/__init__.py [hunk 3]
```diff
@@ -48,6 +48,21 @@ def mjpeg_url(self) -> str:
...
+        return f"{self.base_url}/audio.aac"
+
+    @property
+    def audio_opus_url(self) -> str:
+        """Returns url that LibOPUS audio can be streamed from."""
```
Comment (MartinHjelmare) [inline pydroid_ipcam/__init__.py:63]:
```suggestion
        """Return url that LibOPUS audio can be streamed from."""
```

**Review by MartinHjelmare — APPROVED** [review]
Thanks!

**Pull request merged**